plugins/redis/README

```diff
@@ -0,0 +1,124 @@
+/**********************/
+/*    Redis Plugin    */
+/**********************/
+
+This ECL plugin library is for the persistent key/value store 'redis' (http://redis.io). Utilising the C API
+'hiredis' (http://github.com/redis/hirediscontains) and the event manager library libev
+(http://software.schmorp.de/pkg/libev.html). Dependency notes at bottom.
+
+It contains three modules - 'sync', 'async', and 'locking'. Each one containing the core GET/SET variations
+(by ECL type) differing at their base communication layer with the redis servers. 'async' and 'locking' both
+use the asynchronous hiredis API, and 'sync' the synchronous API. Due to the nature of ECL as a declarative
+and non-functional language, no async activity breaches directly into the ECL layer in a usable manner, i.e.
+callbacks can not be used by the ECL activities EVENT and/OR WAIT. This being said, due to the current
+constraints the ECL plugin infrastructure, some async logic does actually surface into the ECL layer but is
+not directly accessible by any ECL, other than via these plugin functions. Such logic only concerns the
+pub/sub nature of redis based around a locking mechanism on keys.
+
+A full list of redis commands can be found at http://redis.io/commands of which the following are directly
+wrapped in the sync module only:
+
+BOOLEAN Exists(CONST VARSTRING options, CONST VARSTRING key, UNSIGNED database = 0)
+FlushDB(CONST VARSTRING options, UNSIGNED database = 0)
+Del(CONST VARSTRING options, CONST VARSTRING key, UNSIGNED database = 0)
+Persist(CONST VARSTRING options, CONST VARSTRING key, UNSIGNED database = 0)
+Expire(CONST VARSTRING options, CONST VARSTRING key, UNSIGNED4 expire, UNSIGNED database = 0)
+INTEGER DBSize(CONST VARSTRING options, UNSIGNED database = 0)
+
+The redis command 'INFO ALL' is called once per instance of server:port combination and is written to the log
+file (sync only). Only sync connections to a single redis server:port are cached internally, changes in db are
+handled internally by the redis command 'SELECT DB'.
+
+The core SET functions have the prototype pattern (e.g.)
+SetString(CONST VARSTRING options, CONST VARSTRING key, CONST STRING value, UNSIGNED database = 0,  UNSIGNED4 expire = 0 /*days*/)
```

Comment:
I would remove CONST probably confuses things (and shouldn't really be needed on the prototype.)